1.0.0

JoshBBOXX opened the pull request:
# Changes
- Created AfricasTalkingProvider
- Implemented send functionality

# Tests
- Tests for send success and failure added

# Notes
- Repo should roughly emulate existing provider modules: [smsframework-clickatell], [smsframework-vianett]
- Provider works in conjunction with [smsframework]
- Functionality for premium sending, and receiving incoming messages to be added in future

[smsframework-clickatell]: https://github.com/kolypto/py-smsframework-clickatell
[smsframework-vianett]: https://github.com/kolypto/py-smsframework-vianett
[smsframework]: https://github.com/kolypto/py-smsframework

JoshBBOXX committed:
Added AfricasTalkingProvider class declaration

JoshBBOXX committed:
Added send method to AfricasTalkingProvider

- Created AfricasTalkingProviderError
- Updated README with init, config & sending instructions

JoshBBOXX committed:
Added tests for AfricasTalkingProvider send method

panoskar86 commented on requirements.txt:
```diff
@@ -1,2 +1,3 @@
 africastalking==1.1.5
+phonenumbers==8.10.12
```
I think in DOSH we have the light phonenumbers library. phonenumberslite==8.10.0
Check if we really need the full library

JoshBBOXX replied:
Changed to lite lib

panoskar86 commented on smsframework_africastalking/error.py:
```diff
@@ -0,0 +1,6 @@
+class AfricasTalkingProviderError(ProviderError):
+    
+    def __init__(self, message):
+        super().__init__(message)
```
you need to press enter and add one more empty line(PEP-8) :)

JoshBBOXX replied:
done

panoskar86 commented on smsframework_africastalking/provider.py:
```diff
@@ -0,0 +1,62 @@
+            raise AfricasTalkingProviderError('API_ERROR')
+
+        message.msgid = sent_message['messageId']
+        return message
```
PEP-8 :)

JoshBBOXX replied:
done

panoskar86 commented on smsframework_africastalking/provider.py:
```diff
@@ -0,0 +1,62 @@
+                phonenumbers.PhoneNumberFormat.E164
+            )
+        except:
+            raise AfricasTalkingProviderError('PHONE_NUMBER_PARSE_ERROR')
```
do we need to log anything?

JoshBBOXX replied:
Added some generic logging

sramonsanz reviewed (approved):
Looks great apart from few PEP-8 stuff under `africastalking_test.py`.

panoskar86 commented on tests/africastalking_test.py:
```diff
@@ -0,0 +1,58 @@
+        message_out = OutgoingMessage('+254789789789', 'Hello Kenya',
+            provider='africas_talking'
+        )
+        self.assertRaises(AfricasTalkingProviderError, self.gw.send, message_out)
```
PEP-8

JoshBBOXX replied:
done

panoskar86 reviewed (changes requested):
looks good. some really small changes for PEP-8 and I asked some questions

JoshBBOXX committed:
Requested Changes

- PEP8 compliance
- Added logging to send method
- Replaced phonenumbers module with lite version

JoshBBOXX committed:
Added travis integration, tox and PyPI setup file

panoskar86 reviewed (approved):
:+1:
